src/Stratis.Bitcoin.Features.Consensus/Rules/ProvenHeaderRules/ProvenHeaderCoinstakeRule.cs

```diff
@@ -214,19 +203,28 @@ private void CheckStakeKernelHash(PosRuleContext context, UnspentOutputs staking
         {
             OutPoint prevOut = this.GetPreviousOut(header);
             uint transactionTime = header.Coinstake.Time;
+            uint headerBits = chainedHeader.Header.Bits.ToCompact();
 
-            ChainedHeader prevChainedHeader = chainedHeader.Previous;
+            uint256 previousStakeModifier;
 
-            BlockStake prevBlockStake = this.PosParent.StakeChain.Get(prevChainedHeader.HashBlock);
-            if (prevBlockStake == null)
+            if (chainedHeader.Height == this.ProvenHeadersActivationHeight)
             {
-                this.Logger.LogTrace("(-)[BAD_PREV_STAKE]");
-                ConsensusErrors.PrevStakeNull.Throw();
+                // If we are validating the first Proven Header, we don't have a StakeModifierV2 available and I should compute it but I can't.
+                // Proposed solution is to store the StakeModifierV2 value of the header at the height of ProvenHeaderActivation in a consensus variable
+                // called ProvenHeaderActivationStakeModifier and so I can return it
+                previousStakeModifier = uint256.Zero; //TODO to be changed after ProvenHeaderActivationStakeModifier has been created.
```

Comment:
Here you need to get Stakemodifier from store (and when we use checkpoints it will be form the last checkpoint, so perhaps throw an exception for now.

Reply:
yes we have to remove ProvenHeadersActivationHeight too (this will impact almost all PH code)